scipy/signal/_fir_filter_design.py

```diff
@@ -622,30 +634,34 @@ def firwin2(numtaps, freq, gain, *, nfreqs=None, window='hamming',
     if nfreqs is None:
         nfreqs = 1 + 2 ** int(ceil(log(numtaps, 2)))
 
-    if (d == 0).any():
+    if xp.any(d == 0):
         # Tweak any repeated values in freq so that interp works.
-        freq = np.array(freq, copy=True)
-        eps = np.finfo(float).eps * nyq
-        for k in range(len(freq) - 1):
+        freq = xp.asarray(freq, copy=True)
+        eps = xp.finfo(xp_default_dtype(xp)).eps * nyq
```

Comment:
Seem odd that this isn't using `freq.dtype` but it matches the original so :shrug:

Reply:
Agreed on both counts :-). And indeed, matching the original is what is meant to be in scope here.